lib/elixir/test/elixir/kernel/expansion_test.exs

```diff
@@ -1180,6 +1180,16 @@ defmodule Kernel.ExpansionTest do
     assert_raise CompileError, ~r"unhandled operator ->", fn ->
       expand(quote do: (foo -> bar))
     end
+
+    assert_raise CompileError,
```

Comment:
So, if I understood correctly, it's better to revert the last commit. It's done :)

The warning will be suppressed by the `expand/1`, but we don't check it, only that the `CompilationError` happens (before the code it would cause a `CaseClauseError`).